Jellyfin.Networking/Manager/NetworkManager.cs

```diff
@@ -179,409 +135,356 @@ public void Dispose()
+                    _logger.LogWarning("No interface information available. Using loopback interface(s).");
 
-                // No bind address and no exclusions, so listen on all interfaces.
-                Collection<IPObject> result = new Collection<IPObject>();
+                    if (IsIpv4Enabled && !IsIpv6Enabled)
+                    {
+                        _interfaces.Add(new IPData(IPAddress.Loopback, new IPNetwork(IPAddress.Loopback, 8), "lo"));
+                    }
 
-                if (IsIP6Enabled && IsIP4Enabled)
-                {
-                    // Kestrel source code shows it uses Sockets.DualMode - so this also covers IPAddress.Any
-                    result.AddItem(IPAddress.IPv6Any);
-                }
-                else if (IsIP4Enabled)
-                {
-                    result.AddItem(IPAddress.Any);
-                }
-                else if (IsIP6Enabled)
-                {
-                    // Cannot use IPv6Any as Kestrel will bind to IPv4 addresses.
-                    foreach (var iface in _interfaceAddresses)
+                    if (!IsIpv4Enabled && IsIpv6Enabled)
                     {
-                        if (iface.AddressFamily == AddressFamily.InterNetworkV6)
-                        {
-                            result.AddItem(iface.Address);
-                        }
+                        _interfaces.Add(new IPData(IPAddress.IPv6Loopback, new IPNetwork(IPAddress.IPv6Loopback, 128), "lo"));
                     }
                 }
 
-                return result;
+                _logger.LogDebug("Discovered {0} interfaces.", _interfaces.Count);
+                _logger.LogDebug("Interfaces addresses: {0}", _interfaces.OrderByDescending(s => s.AddressFamily == AddressFamily.InterNetwork).Select(s => s.Address.ToString()));
             }
-
-            // Remove any excluded bind interfaces.
-            return _bindAddresses.Exclude(_bindExclusions, false);
         }
 
-        /// <inheritdoc/>
-        public string GetBindInterface(string source, out int? port)
+        /// <summary>
+        /// Initialises internal LAN cache.
+        /// </summary>
+        private void InitialiseLan(NetworkConfiguration config)
         {
-            if (!string.IsNullOrEmpty(source) && IPHost.TryParse(source, out IPHost host))
+            lock (_initLock)
             {
-                return GetBindInterface(host, out port);
-            }
+                _logger.LogDebug("Refreshing LAN information.");
 
-            return GetBindInterface(IPHost.None, out port);
-        }
+                // Get configuration options
+                string[] subnets = config.LocalNetworkSubnets;
 
-        /// <inheritdoc/>
-        public string GetBindInterface(IPAddress source, out int? port)
-        {
-            return GetBindInterface(new IPNetAddress(source), out port);
-        }
+                _ = NetworkExtensions.TryParseToSubnets(subnets, out _lanSubnets, false);
```

Comment:
It's generally a bad idea to rely on the out param of a Try-method if it returns false as you usually only guarantee the value when it's true. It would probably be better to assign Array.Empty if it's false.